src/server.py

```diff
@@ -62,13 +67,37 @@ def response_error(code, phrase):
 def parse_request(client_request):
     """Parse info from a request.  Raise Errors is bad request."""
     http_request = Request(client_request)
-
+    # print(http_request.protocol, http_request.http_method, http_request.headers)
     if http_request.http_method != 'GET':
         raise TypeError("This server only accomodates get requests.")
-    if http_request.correct_get_request():
+    elif http_request.protocol != 'HTTP/1.1':
+        raise ValueError("Incorrect Protocol")
+    elif not http_request.has_host:
+        raise ValueError("No Host found.")
+    else:
         return http_request.uri
+
+
+def resolve_uri(uri):
+    # from pdb import set_trace; set_trace()
+    relative_uri = os.path.join(ROOT, uri.lstrip('/'))
+    if os.path.isdir(relative_uri):
+        try:
+            dir_list = os.listdir(relative_uri)
+            response = ('text', dir_list)
+        except OSError:
+            raise OSError("directroy not found")
     else:
-        raise ValueError("Please make a valid request")
+        try:
+            with open(relative_uri, 'r') as target_file:
```

Comment:
What if you are reading an image?  A pdf file?  Those are binary formats that should never need to be treated as text.  In fact, in a Windows environment treating an image file as a text file like this will corrupt the image.